track.c

```diff
@@ -1128,7 +1147,14 @@ struct aircraft *trackUpdateFromMessage(struct modesMessage *mm)
     }
 
     if (mm->callsign_valid && accept_data(&a->callsign_valid, mm->source)) {
+        if (strcmp(a->callsign, mm->callsign) != 0) {
+            // The callsign changed so tell interactive to
+            // re-evaluate its callsign filter regex if it has one
+            a->callsign_matched = 0;
+        }
         memcpy(a->callsign, mm->callsign, sizeof(a->callsign));
+    } else {
+        a->callsign_matched = 0;
```

Comment:
I don't think you want this else branch. Most received messages do not contain a callsign so mm->callsign_valid is false, but that doesn't mean that the stored callsign has been invalidated.

Reply:
Hmmm.  Most messages from the same aircraft?  I didn't know that.    I'll remove it.

Reply:
ADS-B transponders transmit "aircraft identification & category" messages (which carry the callsign) only once per 5 seconds (or once per 10 seconds when stationary on the ground). It's also carried in some Comm-B responses but those depend on SSR interrogation rates.

Meanwhile you might be getting 20+ messages/second of other types from the same aircraft.